src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -46,6 +46,10 @@ export class ToggleTerminalAction extends TogglePanelAction {
 		if (this.terminalService.terminalInstances.length === 0) {
 			// If there is not yet an instance attempt to create it here so that we can suggest a
 			// new shell on Windows (and not do so when the panel is restored on reload).
+			const toDispose = this.terminalService.onInstanceProcessIdReady(() => {
+				this.terminalService.getActiveInstance().focus();
+				toDispose.dispose();
+			});
 			this.terminalService.createInstance(undefined, true);
```

Comment:
I think it would be safer to keep a reference to the ITerminalInstance created here and make sure that it's correct before you focus/dispose the listener